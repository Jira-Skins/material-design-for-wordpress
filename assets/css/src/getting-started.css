@import "@material/typography/dist/mdc.typography.css";
@import "@material/layout-grid/dist/mdc.layout-grid.css";
@import "@material/button/dist/mdc.button.css";
@import "./gsm/index.css";
