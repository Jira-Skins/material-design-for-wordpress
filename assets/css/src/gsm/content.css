.material-gsm__content {
	padding-right: 1rem;
}
