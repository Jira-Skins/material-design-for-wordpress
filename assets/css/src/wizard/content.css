.material-wizard__content {
	background-color: #fff;
	box-shadow: 0 2px 2px rgba(0, 0, 0, 0.15);
	padding: 2.5rem;

	& p {
		font-family: var(--mdc-typography-body1-font-family, var(--mdc-typography-font-family, Roboto,));
		font-size: var(--mdc-typography-body1-font-size, 1rem);
		line-height: var(--mdc-typography-body1-line-height);
		font-weight: var(--mdc-typography-body1-font-weight, 400);
		letter-spacing: var(--mdc-typography-body1-letter-spacing, 0.03125em);
		text-decoration: var(--mdc-typography-body1-text-decoration, inherit);
		text-transform: var(--mdc-typography-body1-text-transform, inherit);
	}

	& hr {
		margin: 2.5rem auto;
		width: 80%;
	}

	& .material-wizard__title {
		margin-top: 0;
	}

	& .material-wizard__card img {
		width: 100%;
	}
}
