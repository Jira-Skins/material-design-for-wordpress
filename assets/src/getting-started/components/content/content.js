export * from './wizard';
export * from './overview';
export * from './customize';
export * from './layout';
