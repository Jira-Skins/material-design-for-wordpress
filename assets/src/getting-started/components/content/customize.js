import { Fragment } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import Button from '../../../wizard/components/navigation/button';

export const Customize = props => {
	return (
		<Fragment>
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Customize your theme', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Set and preview your global theme styles using Material Theme Options in the Customize panel [LINK]. Choose colors, typography, shapes, and icons to express your unique style. The Material Library shows all available Material Blocks. Once you’re satisfied with your changes, hit “Publish” to update your site.',
					'material-theme-builder'
				) }
			</p>
			<p>
				{ __(
					'Jump into the Block Editor for more options.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Starter Styles', 'material-theme-builder' ) }
			</h2>

			<p>
				{ __(
					'Start from our existing styles and use ',
					'material-theme-builder'
				) }
				<a
					href="https://material.io/design/material-theming/overview.html"
					target="_blank"
					rel="noopener noreferrer"
				>
					{ __( 'Material Theming', 'material-theme-builder' ) }
				</a>
				{ __(
					' to create a custom look and feel.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Color Palettes', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Change your primary and secondary colors and see them applied throughout your site. Need help picking colors? Try Material’s ',
					'material-theme-builder'
				) }
				<a
					href="https://material.io/design/color/the-color-system.html#tools-for-picking-colors"
					target="_blank"
					rel="noopener noreferrer"
				>
					{ __( 'color palette generator', 'material-theme-builder' ) }
				</a>
				{ __(
					' to find complementary, analogous, and triadic tonal palettes.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Typography', 'material-theme-builder' ) }
			</h2>

			<p>
				{ __( 'Choose from more than 1,000 typefaces in ', 'material-theme-builder' ) }
				<a
					href="https://fonts.google.com/"
					target="_blank"
					rel="noopener noreferrer"
				>
					{ __( 'Google Fonts', 'material-theme-builder' ) }
				</a>
				{ __(
					' to set your headline and body styles. Want tips on balancing expression with legibility?. Read Material’s ',
					'material-theme-builder'
				) }
				<a
					href="https://material.io/design/typography/the-type-system.html"
					target="_blank"
					rel="noopener noreferrer"
				>
					{ __( 'typography guidelines', 'material-theme-builder' ) }
				</a>
				{ __( ' for inspiration and best practices.', 'material-theme-builder' ) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Corner Styles', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Material Design systematically applies a shape scheme across components. Change the global corner radius for all blocks or individually for each Material block.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Icon Style', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __( 'All ', 'material-theme-builder' ) }
				<a
					href="http://material.io/icons"
					target="_blank"
					rel="noopener noreferrer"
				>
					{ __( 'Material Icons', 'material-theme-builder' ) }
				</a>
				{ __(
					' are included with this plugin. Choose from Filled, Outlined, Rounded, Two-tone, or Sharp styles.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Set Layout & Navigation', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Select how posts display – as either cards or image lists, in a wide or normal width. Additional card display options include raised or outlined cards, and show or hide post comments, author, excerpt, and date.',
					'material-theme-builder'
				) }
				{ __(
					'Choose how comment fields display within each post.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Top app bar', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site from within the top app bar. By default, the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar visible at all times.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Footer', 'material-theme-builder' ) }
			</h2>
			<p>
				{ __(
					'Add footer text and give your users a quick way to jump back to the top of the page. Looking to add widgets at the bottom of your site? Use the Widgets[LINK] settings.',
					'material-theme-builder'
				) }
			</p>

			<div style={ { height: '20px' } }></div>
		</Fragment>
	);
};
