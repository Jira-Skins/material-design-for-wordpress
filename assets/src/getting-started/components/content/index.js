const Content = () => {
	return (
		<div className="material-gsm__content mdc-layout-grid__cell mdc-layout-grid__cell--span-9">
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				Onboarding Wizard
			</h2>
			<p className="material-gsm__content-description">
				Curabitur mauris erat congue posuere lacus torquent et convallis,
				parturient rutrum cras senectus augue penatibus malesuada rhoncus,
				ultricies finibus pretium arcu sagittis aliquet nisi.
			</p>
			<div className="material-gsm__content-actions"></div>
		</div>
	);
};

export default Content;
