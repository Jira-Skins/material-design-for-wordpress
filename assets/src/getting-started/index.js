/**
 * WordPress dependencies
 */
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import GettingStarted from './components';

render( <GettingStarted />, document.getElementById( 'material-gsm' ) );
