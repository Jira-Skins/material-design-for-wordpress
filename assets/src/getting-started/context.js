/**
 * Setups context to be used across the app
 *
 */
import { useReducer, createContext } from '@wordpress/element';
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';
import { TABS } from './constants';

const TabContext = createContext();

const { Provider } = TabContext;

const tabs = Object.keys( TABS );

/**
 * Default state of the world
 *
 */
const initialState = {
	activeTab: 'THEME',
	completed: [ 'WIZARD' ],
	status: STATUS.IDLE,
	actionToInstall: null,
	error: {},
	themeStatus: 'install',
	contentStatus: 'install',
	tabs,
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const TabProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default TabContext;
