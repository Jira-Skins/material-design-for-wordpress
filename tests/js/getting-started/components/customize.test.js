/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Customize } from '../../../../assets/src/getting-started/components/content/customize';
import { TabProvider } from '../../../../assets/src/getting-started/context';

const setup = () => {
	return render(
		<TabProvider>
			<Customize />
		</TabProvider>
	);
};

describe( 'GSM: Customize', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup();
		expect( wrapper ).toMatchSnapshot();
	} );
} );
