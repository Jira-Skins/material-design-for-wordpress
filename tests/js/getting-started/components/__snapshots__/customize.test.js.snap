// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSM: Customize matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        How to edit your Material Theme
      </h2>
      <p
        class="material-gsm__content-description"
      >
        See your global styles applied in real time by customizing your Material Theme Options directly from the Customize panel. The live preview allows you to see your changes applied directly to your site and our Material Library shows all available Material Blocks. Once you’re satisfied with your changes “Publish” the changes to update your site
      </p>
      <p
        class="material-gsm__content-description"
      >
        If you want to override a specific component (like a button) that can be done from the block settings.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Starter Styles
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Start from our existing styles and take advantage of 
        <a
          href="https://material.io/design/material-theming/overview.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Material Theming
        </a>
         by customizing your theme to truly make your site your own.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Color Palettes
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Change your primary and secondary colors and see your site come to life with Material Design’s color system. Need some help? Try Material’s 
        <a
          href="https://material.io/design/color/the-color-system.html#tools-for-picking-colors"
          rel="noopener noreferrer"
          target="_blank"
        >
          color palette generator
        </a>
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Typography (Font Styles)
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Choose from the full set of 
        <a
          href="https://fonts.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Google Fonts.
        </a>
         Headlines are a great place to inject expression. To retain legibility, utilitarian fonts are better for smaller text sizes. Read Material’s 
        <a
          href="https://material.io/design/typography/the-type-system.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          type system
        </a>
          for more guidance.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Shape Size (Corner Styles)
      </h2>
      <p
        class="material-gsm__content-description"
      >
        The Material Shape System enables you to systematically apply shapes across a variety of components. Change the global corner radius for all blocks or individually for each Material block.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Icon Style
      </h2>
      <p
        class="material-gsm__content-description"
      >
        All of the 
        <a
          href="http://material.io/icons"
          rel="noopener noreferrer"
          target="_blank"
        >
          Material Icons
        </a>
         and their variants are included with this plugin. Choose from Filled, Outlined, Rounded, Two-tone, or Sharp icons that best reflect your style.
      </p>
      <div
        class="material-gsm__content-actions"
      >
        <button
          class="mdc-button mdc-button--raised"
        >
          <span
            class="mdc-button__label"
          >
            Next
          </span>
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon trailing-icon"
          >
            navigate_next
          </i>
        </button>
      </div>
      <div
        style="height: 20px;"
      />
    </div>
  </body>,
  "container": <div>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      How to edit your Material Theme
    </h2>
    <p
      class="material-gsm__content-description"
    >
      See your global styles applied in real time by customizing your Material Theme Options directly from the Customize panel. The live preview allows you to see your changes applied directly to your site and our Material Library shows all available Material Blocks. Once you’re satisfied with your changes “Publish” the changes to update your site
    </p>
    <p
      class="material-gsm__content-description"
    >
      If you want to override a specific component (like a button) that can be done from the block settings.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Starter Styles
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Start from our existing styles and take advantage of 
      <a
        href="https://material.io/design/material-theming/overview.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        Material Theming
      </a>
       by customizing your theme to truly make your site your own.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Color Palettes
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Change your primary and secondary colors and see your site come to life with Material Design’s color system. Need some help? Try Material’s 
      <a
        href="https://material.io/design/color/the-color-system.html#tools-for-picking-colors"
        rel="noopener noreferrer"
        target="_blank"
      >
        color palette generator
      </a>
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Typography (Font Styles)
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Choose from the full set of 
      <a
        href="https://fonts.google.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Google Fonts.
      </a>
       Headlines are a great place to inject expression. To retain legibility, utilitarian fonts are better for smaller text sizes. Read Material’s 
      <a
        href="https://material.io/design/typography/the-type-system.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        type system
      </a>
        for more guidance.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Shape Size (Corner Styles)
    </h2>
    <p
      class="material-gsm__content-description"
    >
      The Material Shape System enables you to systematically apply shapes across a variety of components. Change the global corner radius for all blocks or individually for each Material block.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Icon Style
    </h2>
    <p
      class="material-gsm__content-description"
    >
      All of the 
      <a
        href="http://material.io/icons"
        rel="noopener noreferrer"
        target="_blank"
      >
        Material Icons
      </a>
       and their variants are included with this plugin. Choose from Filled, Outlined, Rounded, Two-tone, or Sharp icons that best reflect your style.
    </p>
    <div
      class="material-gsm__content-actions"
    >
      <button
        class="mdc-button mdc-button--raised"
      >
        <span
          class="mdc-button__label"
        >
          Next
        </span>
        <i
          aria-hidden="true"
          class="material-icons mdc-button__icon trailing-icon"
        >
          navigate_next
        </i>
      </button>
    </div>
    <div
      style="height: 20px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
